management_api_app/db/repositories/workspaces.py

```diff
@@ -1,43 +1,60 @@
+import uuid
 from typing import List
 
-from azure.cosmos import ContainerProxy, CosmosClient, PartitionKey
+from azure.cosmos import ContainerProxy, CosmosClient
 
 from core.config import STATE_STORE_RESOURCES_CONTAINER
-from db.errors import EntityDoesNotExist
+from db.errors import EntityDoesNotExist, UnableToAccessDatabase
 from db.repositories.base import BaseRepository
-from models.domain.resource import Resource
+from models.domain.resource import Resource, ResourceType, Status
+from models.schemas.resource import ResourceInCreate
 from resources import strings
 
 
 class WorkspaceRepository(BaseRepository):
     def __init__(self, client: CosmosClient):
-        super().__init__(client)
-        if self.database:
-            self._container = self.database.create_container_if_not_exists(id=STATE_STORE_RESOURCES_CONTAINER, partition_key=PartitionKey(path="/appId"))
-        else:
-            self._container = None
+        super().__init__(client, STATE_STORE_RESOURCES_CONTAINER)
 
     @property
     def container(self) -> ContainerProxy:
         return self._container
 
     def get_all_active_workspaces(self) -> List[Resource]:
```

Comment:
potentially dangerous function - if there are lots of workspaces loading them all in memory could kill the server. IS there no pagination support?

Reply:
@sachinkundu if you are talking about get_all_active_workspaces - that was done in a PR that is already merged... we can create another issue about adding pagination if needed, but would prefer keeping that out of this POST PR.

Reply:
Created issue #139 so that we can prioritize this separately